docs/CONNECTOR_PROTOCOL.md

```diff
@@ -40,6 +40,34 @@ This is our main communication index, used to communicate the connector's config
                            fields
   description: string;  -> the description of the connector
   error: string;        -> Optional error message
+  filtering: {          -> Filtering rules
+    domain: string,     -> what data domain these rules apply to
+    active: {           -> "active" rules are run in jobs. 
+      rules: {
+        id: string,         -> rule identifier
+        policy: string,     -> one of ["include", "exclude"]
+        field: string,      -> the field on the document this rule applies to
+        rule: string,       -> one of ["regex", "starts_with", "ends_with", "contains", "equals", "<", ">"]
+        value: string,      -> paired with the `rule`, this `value` either matches the contents of the document's `field` or does not
+        order: number,      -> the order in which to match rules. The first rule to match has its `policy` applied
+        created_at: string, -> when the rule was added
+        updated_at: string  -> when the rule was last edited
+      },
+      advanced_snippet: {   -> used for filtering data from the 3rd party at query time
+        value: object,      -> this JSON object is passed directly to the connector
+        created_at: string, -> when this JSON object was created
+        updated_at: string  -> when this JSON object was last edited
+      },
+      validation: {
+        state: string,      -> one of ["edited", "valid", "invalid"]
+        errors: {
+          ids: string,      -> the id(s) of any rules that are deemed invalid
```

Comment:
According to the [implementation doc](https://docs.google.com/document/d/1llPDGzduj8lST0N45v_G4PB0s0RUOBFurOIsBAD-Fyg/edit#heading=h.l2heyce8rz9r), `errors` is an array, and I see an object here.